Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Managers/VaultManager.swift

```diff
@@ -0,0 +1,824 @@
+//
+//  VaultManager.swift
+//  PrimerSDK
+//
+//  Created by Evangelos Pittas on 13/6/23.
+//
+
+#if canImport(UIKit)
+
+import SafariServices
+import UIKit
+
+extension PrimerHeadlessUniversalCheckout {
+    
+    public class VaultManager: NSObject {
+        
+        private(set) var vaultedPaymentMethods: [PrimerHeadlessUniversalCheckout.VaultedPaymentMethod]?
+        private(set) var paymentMethodTokenData: PrimerPaymentMethodTokenData?
+        private(set) var paymentCheckoutData: PrimerCheckoutData?
+        private(set) var resumePaymentId: String?
+        private var webViewController: SFSafariViewController?
+        private var webViewCompletion: ((_ authorizationToken: String?, _ error: PrimerError?) -> Void)?
+        
+        public override init() {
+            PrimerInternal.shared.sdkIntegrationType = .headless
+            PrimerInternal.shared.intent = .checkout
+            
+            let sdkEvent = Analytics.Event(
+                eventType: .sdkEvent,
+                properties: SDKEventProperties(
+                    name: "\(Self.self).\(#function)",
+                    params: nil))
+            Analytics.Service.record(events: [sdkEvent])
+            
+            super.init()
+        }
+        
+        public func configure() throws {
+            try self.validate()
+        }
```

Comment:
This function is not needed. init can throw and we can handle the error during VaultManager initialisation.